hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java

```diff
@@ -124,12 +132,55 @@ public Fees dispatchComputeFees(@NonNull final FeeContext feeContext) {
 
         try {
             final var handler = getHandler(feeContext.body());
-            return handler.calculateFees(feeContext);
+            var fee = handler.calculateFees(feeContext);
+            return scale(feeContext.body(), fee);
         } catch (UnsupportedOperationException ex) {
             throw new HandleException(ResponseCodeEnum.INVALID_TRANSACTION_BODY);
         }
     }
 
+    // Scaling the fees if certain conditions are met.
+    // E.g. when we do a ScheduleCreate we want to scale the fees based on what transaction type is scheduled
+    private static Fees scale(TransactionBody transactionBody, Fees fees) {
+        int multiplier = 1; // by default, we are not scaling
+
+        final var functionality = transactionBody.data().kind();
+        // We are only scaling the ScheduleCreate. In the future we can add more functionalities.
+        switch (functionality) {
+            case SCHEDULE_CREATE -> multiplier = getScheduleMultiplier(transactionBody);
+        }
+
+        return fees.multiplyBy(multiplier);
+    }
+
+    // Getting the scheduled multiplier based on what transaction type is scheduled
+    private static int getScheduleMultiplier(TransactionBody transactionBody) {
+        try {
+            HederaFunctionality functionality = getScheduledHederaFunctionality(transactionBody);
+            var multiplier =
+                    new AssetsLoader().loadScheduledTransactionMultipliers().get(functionality);
```

Comment:
Is this reading a file from disk on the handle thread every time we encounter a ScheduleCreate?